tests/unit/test_util.py

```diff
@@ -37,32 +37,46 @@ class FakeModel(FakeApiModelExtension):
     name: str
 
 
+class FakeModel2(FakeApiModelExtension):
+    code: int = 0
+    name: str
+    status: str
+
+
 class TestParseInput:
 
     def test_no_data(self, app):
         """
         GIVEN a request with no parameters of any type
         WHEN we call parse_input on that request
-        THEN we get a BaseApiModel returned with a code of 400
+        THEN we get a BaseApiView returned with a code of 400
         AND an error explaining what is wrong
         """
         with app.test_request_context():
             retval = parse_input(app.logger, FakeModel)
-            assert 'Field required' in retval.error
-            assert isinstance(retval, BaseApiModel)
+            assert (len(retval.errors)) == 1
+            assert 'name' == [*retval.errors][0]
+            assert 'Field required' == retval.errors.get('name')
+            assert isinstance(retval, BaseApiView)
 
-    def test_no_data_custom_return(self, app):
+    @pytest.mark.parametrize("to_parse,attrs_to_failed",
+                             [(FakeModel, ['name']),
+                              (FakeModel2, ['name', 'status'])])
+    def test_no_data_custom_return(self, to_parse, attrs_to_failed, app):
         """
         GIVEN a request with no parameters of any type
         WHEN we call parse_input on that request with a custom default_return
         THEN we get a custom object returned with additional fields
         AND an error explaining what is wrong
         """
         with app.test_request_context():
-            retval = parse_input(app.logger, FakeModel, FakeApiModelExtension)
+            retval = parse_input(app.logger, to_parse, FakeApiModelExtension)
             assert retval.code == 1
-            assert 'Field required' in retval.error
-            assert isinstance(retval, BaseApiModel)
+            assert retval.msg == f"Errors detected: {len(attrs_to_failed)}"
+            for missing_attr in retval.errors:
+                assert missing_attr in attrs_to_failed
```

Comment:
this looks like what I was suggesting above!

Reply:
I've noticed now! :smiley: